common/logger/src/sample.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Implement periodic sampling for logs
+
+use std::{
+    sync::atomic::{AtomicU64, Ordering},
+    time::{Duration, SystemTime},
+};
+
+#[derive(Debug)]
+pub enum SampleRate {
+    Duration(Duration),
+    Frequency(u64),
+    Always,
```

Comment:
I would add some comments around this.  Mostly around Frequency e.g. `Frequency(0)` or `Frequency(1)` is the same as `Always`.  `Frequency(2)` is every other message.  We should reject `Frequency(0)` or `Frequency(1)` and have users input `Always`, though I'm not sure we need an `Always`.

Reply:
I agree that we should reject such inputs, unfortunately we probably won't be able to do so until we can do just a bit more in const functions.